### ISSUE TITLE
Issues with reprs

### ISSUE THREAD

**Issue by ceball** [ioam/paramnb#55]
I will split this issue up into smaller issues describing what I think is best to do; just noting various issues for now...

For this example code:

```
import param
import paramnb
```
```
class Something(param.Parameterized):
    a = param.Number(default=1)

s = Something()
```
```
class SomeFn(param.ParameterizedFunction):
    a_dict = param.Dict(default={'s':s})
    a_list = param.List(default=[s])
    a_param = param.Parameter(default=s)
    a_selector = param.ObjectSelector(objects=[s])

f = SomeFn.instance()
```
```
paramnb.Widgets(f)
```

## Inconsistent reprs

<img width="436" alt="screen shot 2017-06-27 at 12 37 54 pm" src="https://user-images.githubusercontent.com/1929/27585784-80f1f92e-5b35-11e7-9dd8-6b479ff6f89b.png">

## Confusion about editing

![screencast 2017-06-27 12-39-28](https://user-images.githubusercontent.com/1929/27585905-efd16ef6-5b35-11e7-90ab-954f6a42557f.gif)

(Note re. above: it's not clear because of the loop, but once it's gone orange there's no way back except recreating the widgets.)

![screencast 2017-06-27 12-55-22](https://user-images.githubusercontent.com/1929/27586424-2fdbb824-5b38-11e7-97a6-8239cf12e3ba.gif)

**Comment by philippjfr** [comment]
Yeah not sure what to do here yet, the text boxes are set up to handle python literals, parameterized objects weren't considered.
